kore/src/Kore/Internal/MultiOr.hs

```diff
@@ -361,10 +357,35 @@ observeAllT :: (Ord a, TopBottom a, Monad m) => LogicT m a -> m (MultiOr a)
 observeAllT act = make <$> Logic.observeAllT act
 {-# INLINE observeAllT #-}
 
+observeAll :: (Ord a, TopBottom a) => Logic a -> MultiOr a
+observeAll = make . Logic.observeAll
+{-# INLINE observeAll #-}
+
 map
     :: Ord child2
     => TopBottom child2
     => (child1 -> child2)
     -> MultiOr child1
     -> MultiOr child2
 map f = make . fmap f . extractPatterns
+{-# INLINE map #-}
+
+traverse
+    :: Ord child2
+    => TopBottom child2
+    => Applicative f
+    => (child1 -> f child2)
+    -> MultiOr child1
+    -> f (MultiOr child2)
+traverse f = fmap make . Traversable.traverse f . extractPatterns
+{-# INLINE traverse #-}
+
+traverseLogic
+    :: Ord child2
+    => TopBottom child2
+    => Monad m
+    => (child1 -> m child2)
+    -> MultiOr child1
+    -> m (MultiOr child2)
+traverseLogic f multiOr = observeAllT $ Logic.scatter multiOr >>= lift . f
+{-# INLINE traverseLogic #-}
```

Comment:
Why do we need this? It seems exactly the same as `traverse`.

Reply:
I didn't end up using it, but left it in case anyone would want to run effects with `LogicT` instead of using the `traverse` instance on lists.

Reply:
But, there's no reason to do that. The only reason to use `LogicT` is if you want to run multiple traversals in sequence and build up the results lazily. There's no reason to use `LogicT` here because we immediately call `observeAllT`.

By the way, I just remembered: `traverse` should have a prominent warning that it should _not_ be used with `LogicT`.